Implemented CRUD features

DouHoonKwark opened the pull request:
1. Implemented CRUD features based on our ideas.
2. Still need to improve image src and some constraints of input
3. Also need to polish CSS and JS files

adaburton committed:
Created the BookEntries JS file, and added an edit button.

adaburton committed:
Added book list webcomponent, and modified bookentries for webcomponent

Zweber01 committed:
Updated code

adaburton committed:
Implemented the add list button, add book button, and delete book button

amaansingh160 committed:
added lists to local, need to add books objects

adaburton committed:
Merge branch 'main' into bookentries-JS

DouHoonKwark committed:
Update BookEntry.js

Found a minor syntax error(?) that prevented the add-book button from working properly.

DouHoonKwark committed:
Update bookEntries.html

found the reason why the weird // word on the top leftmost of the book entries page occurred and deleted it

DouHoonKwark committed:
deleted ===== in some files

deleted ===== that prevented the source file from working properly

adaburton committed:
Started implementing the dialog popup for adding a book

adaburton committed:
Merge branch 'bookentries-JS' of https://github.com/cse110-fa22-group13/cse110-fa22-group13 into bookentries-JS

adaburton committed:
Added edit button functionality

adaburton committed:
dialog box shows when add-book button is clicked

adaburton committed:
bookcover edit button shows the dialog box

adaburton committed:
book is no longer added if the add button is not pressed

adaburton committed:
dialog box disappears if user clicks outside of it

adaburton committed:
nothing, just added spaces

DouHoonKwark committed:
implemented local storage for add button

DouHoonKwark committed:
Implemented CRUD features

Finished implementing UD features that have not been implemented. The key idea is to generate a unique value for each entry and list so that we can easily distinguish the specific item in the local storage.

DouHoonKwark committed:
small updates including css style

DouHoonKwark committed:
form gets cleared when the modal is unexpectedly closed

DouHoonKwark committed:
Merge branch 'bookentries-JS' into 11-30-22-bookentries

adaburton committed:
Merge pull request #145 from cse110-fa22-group13/11-30-22-bookentries

Implemented features and some minor changes

DouHoonKwark committed:
Merge branch 'main' into bookentries-JS

Donttestturing committed:
Add testing environment and first tests

Donttestturing reviewed (changes requested):
Need to Fix any html validator and linting issues before merging

DouHoonKwark committed:
Polish the basic CRUD features and CSS

1. Implemented the delete button for each entry list and its functionality.
2. CSS for modal and entry list
3. default settings based one what behavior a user decides to do

Donttestturing committed:
Add rule for linter to ignore jsdocs

Donttestturing committed:
add back jsdoc require for linter, remove jsdoc syntax errors

DouHoonKwark committed:
fixed the HTML issues that HTML validator complained

fixed all HTML issues that the validator complained

DouHoonKwark committed:
Merge branch 'bookentries-JS' of https://github.com/cse110-fa22-group13/cse110-fa22-group13 into bookentries-JS

Donttestturing committed:
Small change to text of form

adaburton committed:
JSDocs

adaburton committed:
More JSDocs

adaburton committed:
CSS Style, and minor fixes

adaburton committed:
more CSS styling

adaburton committed:
Even more CSS Style

adaburton committed:
css styling

adaburton committed:
css style

adaburton committed:
hopefully the last css style update for a while

DouHoonKwark commented:
It all passed now. Thank you !

Donttestturing reviewed (approved):
Cool! We have the CRUD functionality working, as well as tests. Thank y'all for adding the Documentation too! Though that can still be improved. Thanks everybody!
